bot/exts/valentines/markov_poem_generator.py

```diff
@@ -0,0 +1,512 @@
+                lines.append("")
+                continue
+
+            # Creating a line for the unit
+            if unit not in rhyme_track:
+                new_line = await self._init_unit(
+                    rhyme_track, unit, is_last_line
+                )
+                lines.append(new_line)
+            else:
+                new_line = await self._get_rhyming_line(
+                    word_rhymes=rhyme_track[unit],
+                    existing_lines=lines,
+                    error_info=(ctx, is_first_error, scheme, unit_count,
+                                time_start)
+                )
+                lines.append(new_line)
+
+                # If last line, it will not be referred to again
+                if not is_last_line:
+                    rhyme_track[unit] |= await self._get_rhyme_set(
+                        instance=self,
+                        word=self._get_last_word(new_line)
+                    )
+
+            unit_count[unit] -= 1
+
+        return lines
+
+    async def send_markov_poem(
+        self,
+        ctx: commands.Context,
+        scheme: str,
+        unit_count: Dict[str, int],
+        time_start: datetime.time,
+        is_first_error: bool = True
+    ) -> None:
+        """Gives the user a paginated embed of their markov poem."""
+        lines = await self._get_generated_markov_poem(
+            instance=self,
+            timeout=self.POEM_TIMEOUT,
+            ctx=ctx,
+            scheme=scheme,
+            unit_count=unit_count,
+            time_start=time_start,
+            is_first_error=is_first_error
+        )
+
+        elapsed_time = datetime.now() - time_start
+        elapsed_time = elapsed_time.seconds
+
+        poem_embed = Embed(
+            title="A Markov Poem For " + str(ctx.author.name),
+            color=Colours.soft_red
+        )
+
+        await LinePaginator.paginate(
+            lines,
+            ctx,
+            poem_embed,
+            max_lines=31,
+            max_size=1000,
+            empty=False,
+            footer_text=f"Elapsed time: {elapsed_time}s\n"
+                        "Rhymes API provided by datamuse."
+        )
+
+    @commands.max_concurrency(1, commands.BucketType.user)
+    @commands.command(
+        aliases=("poem", "mpoem", "m_poem", "markpoem", "mark_poem"),
+        help=f"""This command generates a poem via a markov chain, and all you
+            have to do is provide a rhyme scheme.
+
+            The rhyme scheme is made from characters separated by slashes. E.g
+            "abab/cdcd/efef/gg". The slashes denote a new stanza, i.e, they
+            create an empty line. Same characters mean that the lines rhyme.
+            Note that the characters are case sensitive! For example, a and A
+            represent two different rhyme schemes.
+
+            You may also use our existing rhyme scheme templates:
+            {chr(10).join((k + " --- " + v) for k, v in templates.items())}""",
```

Comment:
Do we want users to see this?

Reply:
Yep, so that the help command displays the possible traditional rhyme schemes. Here is what it looks like:
![image](https://user-images.githubusercontent.com/32971425/115367029-65fd6a00-a1f8-11eb-827c-f337ec331e00.png)